From b04ab8dfbed72b0bbd277a8533e7c608e16488c1 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Fri, 6 Mar 2020 11:16:50 +0100
Subject: [PATCH 3/4] tqma6ul: check fuse state for ENET1/2 on MBa6ULxL

Since there are CPU variants which fuses ENET IP, check before access.
Use tested code from MBa6ULx

While at it, fix two typos

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 board/tqc/tqma6ul/tqma6ul_mba6ulxl.c | 40 +++++++++++++++++-----------
 1 file changed, 25 insertions(+), 15 deletions(-)

diff --git a/board/tqc/tqma6ul/tqma6ul_mba6ulxl.c b/board/tqc/tqma6ul/tqma6ul_mba6ulxl.c
index 51aec32e1fd..02c71293e54 100644
--- a/board/tqc/tqma6ul/tqma6ul_mba6ulxl.c
+++ b/board/tqc/tqma6ul/tqma6ul_mba6ulxl.c
@@ -145,10 +145,10 @@ static void mba6ulxl_setup_iomuxc_enet(void)
 
 int board_eth_init(bd_t *bis)
 {
-	int ret;
+	int ret = 0;
 
 	/*
-	 * FEC0 and FEC1 shares the mdio bus therefore the
+	 * FEC0 and FEC1 shares the mdio bus. Therefore the
 	 * CONFIG_FEC_MXC_MDIO_BASE macro is used to specify the bus.
 	 * This makes the ENET_BASE_ADDR and ENET2_BASE_ADDR macro
 	 * useless for the fecmxc_initialize_multi function.
@@ -156,15 +156,22 @@ int board_eth_init(bd_t *bis)
 	 * correct fused MAC address - renaming can be done under linux
 	 * using udev / systemd
 	 */
-	ret = fecmxc_initialize_multi(bis, 0, TQMA6UL_ENET1_PHYADDR,
-				      ENET_BASE_ADDR);
-	if (ret)
-		printf("FEC0 MXC: %s:failed %i\n", __func__, ret);
-
-	ret = fecmxc_initialize_multi(bis, 1, TQMA6UL_ENET2_PHYADDR,
-				      ENET2_BASE_ADDR);
-	if (ret)
-		printf("FEC1 MXC: %s:failed %i\n", __func__, ret);
+	if (check_module_fused(MX6_MODULE_ENET1)) {
+		puts("FEC0: fused\n");
+	} else {
+		ret = fecmxc_initialize_multi(bis, 0, TQMA6UL_ENET1_PHYADDR,
+					      ENET_BASE_ADDR);
+		if (ret)
+			printf("FEC0 MXC: %s:failed %i\n", __func__, ret);
+	}
+	if (check_module_fused(MX6_MODULE_ENET2)) {
+		puts("FEC1: fused\n");
+	} else {
+		ret = fecmxc_initialize_multi(bis, 1, TQMA6UL_ENET2_PHYADDR,
+					      ENET2_BASE_ADDR);
+		if (ret)
+			printf("FEC1 MXC: %s:failed %i\n", __func__, ret);
+	}
 
 	return ret;
 }
@@ -182,9 +189,10 @@ static int mba6ulxl_setup_fec(int fec_id)
 
 	switch (fec_id) {
 	case FEC0:
-		if (check_module_fused(MX6_MODULE_ENET1))
+		if (check_module_fused(MX6_MODULE_ENET1)) {
+			puts("FEC0: fused\n");
 			return -1;
-
+		}
 		/*
 		 * Use 50M anatop loopback REF_CLK1 for ENET1,
 		 * clear gpr1[13], set gpr1[17]
@@ -193,11 +201,13 @@ static int mba6ulxl_setup_fec(int fec_id)
 				IOMUX_GPR1_FEC1_CLOCK_MUX1_SEL_MASK);
 		break;
 	case FEC1:
-		if (check_module_fused(MX6_MODULE_ENET2))
+		if (check_module_fused(MX6_MODULE_ENET2)) {
+			puts("FEC1: fused\n");
 			return -1;
+		}
 		/*
 		 * Use 50M anatop loopback REF_CLK1 for ENET2,
-		 * clear gpr1[13], set gpr1[17]
+		 * clear gpr1[14], set gpr1[18]
 		 */
 		clrsetbits_le32(&iomuxc_gpr_regs->gpr[1], IOMUX_GPR1_FEC2_MASK,
 				IOMUX_GPR1_FEC2_CLOCK_MUX1_SEL_MASK);
-- 
2.17.1

